src/sql/src/plan/statement/show.rs

```diff
@@ -632,31 +708,55 @@ pub fn show_clusters<'a>(
     filter: Option<ShowStatementFilter<Aug>>,
 ) -> Result<ShowSelect<'a>, PlanError> {
     let query = "
-SELECT
-    mc.name,
-    pg_catalog.string_agg(mcr.name || ' (' || mcr.size || ')', ', ' ORDER BY mcr.name)
-        AS replicas
-FROM
-    mz_catalog.mz_clusters mc
-        LEFT JOIN mz_catalog.mz_cluster_replicas mcr ON mc.id = mcr.cluster_id
-GROUP BY mc.name"
-        .to_string();
-    ShowSelect::new(scx, query, filter, None, Some(&["name", "replicas"]))
+WITH clusters AS (
+    SELECT
+        mc.id,
+        mc.name,
+        pg_catalog.string_agg(mcr.name || ' (' || mcr.size || ')', ', ' ORDER BY mcr.name)
+            AS replicas
+    FROM mz_catalog.mz_clusters mc
+    LEFT JOIN mz_catalog.mz_cluster_replicas mcr
+      ON mc.id = mcr.cluster_id
+    GROUP BY mc.id, mc.name
+),
+comments AS (
+    SELECT id, comment
+    FROM mz_internal.mz_comments
+    WHERE object_type = 'cluster' OR object_type IS NULL
```

Comment:
Here I think you actually don't want the `IS NULL` predicate because it's evaluated before the join, unlike after the join in the other implementations.

Reply:
Totally, updated!